dashboard/test/seeding/lesson_seeding_test.rb

```diff
@@ -0,0 +1,146 @@
+require 'test_helper'
+
+class LessonSeedingTest < ActiveSupport::TestCase
+  test 'serialize script lessons' do
+    script = create_script_with_lessons(num_lesson_groups: 2, num_lessons_per_group: 2)
+    # Set names / keys for consistent results regardless of DB state
+    script.name = 'bogus-script-1'
+    script.lesson_groups.each_with_index do |lg, i|
+      lg.key = "Bogus Lesson Group #{i + 1}"
+      lg.save!
+    end
+    script.lessons.each_with_index do |l, i|
+      l.key = "Bogus-Lesson-#{i + 1}"
+      l.save!
+    end
+
+    # Set some properties to serialize
+    lesson1, lesson2, lesson3, _ = script.lesson_groups.map(&:lessons).flatten
+    lesson1.overview = 'overview 1'
+    lesson2.visible_after = 'visible_after 2'
+    lesson3.overview = 'overview 3'
+    lesson3.visible_after = 'visible_after 3'
+
+    script.save!
+
+    expected = <<~JSON
+      {
+        "name": "bogus-script-1",
+        "lesson_groups": [
+          {
+            "key": "Bogus Lesson Group 1",
+            "lessons": [
+              {
+                "key": "Bogus-Lesson-1",
+                "properties": {
+                  "overview": "overview 1"
+                }
+              },
+              {
+                "key": "Bogus-Lesson-2",
+                "properties": {
+                  "visible_after": "visible_after 2"
+                }
+              }
+            ]
+          },
+          {
+            "key": "Bogus Lesson Group 2",
+            "lessons": [
+              {
+                "key": "Bogus-Lesson-3",
+                "properties": {
+                  "overview": "overview 3",
+                  "visible_after": "visible_after 3"
+                }
+              },
+              {
+                "key": "Bogus-Lesson-4",
+                "properties": {
+                }
+              }
+            ]
+          }
+        ]
+      }
+    JSON
+
+    assert_equal expected.strip, LessonSeeding.serialize_lessons(script)
```

Comment:
can we please include a trailing newline in the json file output (not necessarily in this PR / method)?

Reply:
Done, though I'm curious what the reason is.

Reply:
1. it is part of the "POSIX standard": https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
2. it breaks my tools in the sense that I can't use rubymine or my diff tool to remove the trailing newline after rubymine (perhaps the prettier extension running inside it? I'm not sure) automatically adds the trailing newline. instead I have to use `git checkout`.

if other people use tools which have the opposite problem (can't handle the presence of trailing newlines and/or tools automatically remove trailing newlines) then I'd be happy to discuss further.